webapp/channels/src/components/user_settings/general/user_settings_general.tsx

```diff
@@ -407,12 +442,32 @@ export class UserSettingsGeneralTab extends PureComponent<Props, State> {
     };
 
     submitAttribute = async (settings: string[]) => {
+        const {formatMessage} = this.props.intl;
+
         const attributeID = settings[0];
-        const attributeValue = this.state.customAttributeValues?.[attributeID];
-        if (attributeValue == null) {
+        const attributeField = this.props.customProfileAttributeFields.find((field) => field.id === attributeID);
+        if (attributeField === undefined) {
             return;
         }
+        let attributeValue: string | string[] = this.state.customAttributeValues?.[attributeID];
 
+        if (typeof attributeValue === 'string' && attributeField.attrs && attributeField.attrs.value_type) {
+            if (attributeField.attrs.value_type === 'email') {
+                if (attributeValue !== '' && !isEmail(attributeValue)) {
+                    this.setState({clientError: formatMessage(holders.validEmail), emailError: '', serverError: ''});
+                    return;
+                }
+            }
+            if (attributeField.attrs.value_type === 'url') {
+                if (attributeValue !== '' && !isValidUrl(attributeValue)) {
+                    this.setState({clientError: formatMessage(holders.validUrl), emailError: '', serverError: ''});
+                    return;
+                }
+            }
```

Comment:
When saving a url and it's invalid e.g. `http://goo gle.com`, the backend will return an error and webapp hard-crashes.

> `"Invalid or missing value for field 4g6bkhp4cjfkbqwyyjk3icwkhc: invalid url: parse \"http://goo gle.com\": invalid character \" \" in host name in request body."`

> `Error: Objects are not valid as a React child`

Reply:
https://github.com/mattermost/mattermost/blob/b5a188c5689b6da1d36b913a26b55e2537453702/webapp/channels/src/components/channel_bookmarks/channel_bookmarks_create_modal.tsx#L697 
This function might catch more URL errors before sending. Maybe that function should be moved to `utils/url`?

Reply:
Update the error issue first, then updated to use `validHttpUrl()`